### DISCUSSION TITLE
TypeError: Failed to fetch

### DISCUSSION THREAD

**Discussion by recovery20** [code4recovery/12-step-meeting-list#955] [Q&A] [answered]
Hi,

I've been using this plugin across several sites and for the most part, it's working very well. But on some I'm getting an error - "TypeError: Failed to fetch" - whenever I enable the TSML UI. And when I am using the Legacy UI, the table that displays the meetings is not responsive. Is there something wrong with my site configuration that is preventing these from working properly? 

This is an example of one of the domains that isn't working: https://recoverymeetingsnearme.com/ 

Any help would be greatly appreciated. 

-B

**Comment by joshreisner** [comment]
hey @recovery20 - i can help you with the responsive theme issue, looks like the table is overflowing the bounds on mobile and so some text is getting cut off, right? 

seems like your theme isn't getting along with the TSML style. i find that the error goes away if i add this CSS rule:
```
#tsml #meetings .results table tbody tr td.address, #tsml #meetings .results table tbody tr td.district, #tsml #meetings .results table tbody tr td.location, #tsml #meetings .results table tbody tr td.name, #tsml #meetings .results table tbody tr td.region {
  white-space: normal !important;
}
```

before: 
![before](https://user-images.githubusercontent.com/1551689/196496552-b4848fbb-81fb-4fcd-9482-0089dbe20eef.png)

after:
![after](https://user-images.githubusercontent.com/1551689/196496629-23517661-6d4b-4ba0-8e64-cf0439add663.png)

hard to say what's happening with TSML UI without seeing the issue, but it sounds like it can't load your JSON properly. if you were to enable it briefly we could take a look, or perhaps you could create a new hidden page with the `[tsml_ui]` shortcode on it? it probably won't look good but should give us enough info to diagnose the issue

**Comment by recovery20** [comment]
Thank you for the quick reply! I'll try that CSS rule now. 

I added the shortcode to this page: https://recoverymeetingsnearme.com/meeting-test/

**Reply by joshreisner** [reply]
great thanks. the error i see is:
```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://7mtdd1.p3cdn1.secureserver.net/wp-content/tsml-cache-0e73b65dec.json?1665774288. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 200.
```
sounds like your site is confused about what its URL is supposed to be. try going into WordPress settings and checking that the Site Address and WordPress Address are correct

**Comment by recovery20** [comment]
Thanks for the info! I reached out to my hosting provider because I thought it was odd that it was referencing the temporary domain (https://7mtdd1...) and for some reason that temporary domain wouldn't redirect to the primary. So I deleted that and created a new WP instance and although the new temporary domain redirects to the primary domain, I'm still seeing that error. Is there perhaps something I could add to my htaccess that would prevent the error? 

I did check the site address/WordPress address and both of those were/are correct. 

If this isn't something you can help with, no worries. The plugin still works, but I'd really love to get the TSML UI working :/

**Reply by recovery20** [reply]
This page has the TSML UI enabled: https://recoverymeetingsnearme.com/meetings/

**Reply by joshreisner** [reply]
i'm seeing a slightly different URL now - still not the one you're using. is this GoDaddy website caching?
```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://secureservercdn.net/45.40.155.106/z30.ae1.myftpupload.com/wp-content/tsml-cache-fa1e988c9b.json?1666210709. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 200.
```

**Comment by recovery20** [comment]
Correct - hosting is Managed WordPress through GoDaddy

**Comment by joshreisner** [comment, marked as answer]
Ok so basically what is happening is that GoDaddy is serving your website assets through a CDN which has a different domain from your website, which might be fine except it's also disallowing cross-domain data-loading.

I don't think we have a good workaround for this. I will create a ticket to [adjust these lines](https://github.com/code4recovery/12-step-meeting-list/blob/main/includes/shortcodes.php#L129-L132) to specify a data URL that does not include the domain. But for now you should probably use the legacy UI

**Reply by joshreisner** [reply]
https://github.com/code4recovery/12-step-meeting-list/issues/956

**Reply by recovery20** [reply]
Thank you for your help!
